### PR TITLE
Added: Dark mode for Origine +Origine compact themes

### PR DESCRIPTION
**Pull request by math-GH**
Ref #4515
Ref #2793
Follow up of #4693

This PR shows how easy it is to use the variables to create cool new stuff.

This themes have now a dark mode:
- Origine
- Origine compact

grafik

Changes proposed in this pull request:

- media query that ask for the browsers preferences (CSS)

How to test the feature manually:

1. if you have a dark mode in your browser enabled: you see the dark theme
2. if you have NOT a dark mode in your browser enabled: you see the (default) light theme

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested

### CODE REVIEW

**Comment by math-GH** [comment]
This PR is a draft PR to collect your feedback, that is very welcome

**Comment by Frenzie** [comment]
Sure, why not. Are the colors based on anything in particular?

**Comment by math-GH** [comment]
>  Are the colors based on anything in particular?

I tried to invert the white/grey colors to black/dark grey.
Used the same blue color like in the light theme.
Then I tried to modify the dark colors to keep the character of the Origine theme.

I am open for ideas of another color values.

**Comment by math-GH** [comment]
I had to change a bit the blue color for a better reading on the dark background.

New added:
The new option in the theme configuration:
grafik

File: app/i18n/fr/conf.php [hunk 1]
```diff
@@ -32,6 +32,7 @@
 	),
 	'display' => array(
 		'_' => 'Affichage',
+		'darkMode' => 'Automatically dark mode (Beta)',	// TODO
```
Comment (Alkarex) [inline app/i18n/fr/conf.php]:
```suggestion
		'darkMode' => 'Mode sombre automatique (bêta)',
```

Comment (Frenzie) [reply app/i18n/fr/conf.php]: The original should be `Automatic dark mode (beta)` btw.

Automatic for grammar, beta for style.

File: config-user.default.php [hunk 2]
```diff
@@ -62,6 +62,7 @@
 		'site' => true,
 	),
 	'theme' => 'Origine',
+	'darkMode' => false,
```
Comment (Alkarex) [inline config-user.default.php]: Do we maybe want three states: { yes, no, automatic } ?

Comment (Frenzie) [reply config-user.default.php]: Agreed, probably with automatic as the default?

Comment (math-GH) [reply config-user.default.php]: Maybe later.
`yes` would mean to find another solution than the media query `@media (prefers-color-scheme: dark)`.

Comment (Alkarex) [reply config-user.default.php]: There is a simple approach on https://stackoverflow.com/questions/56300132/how-to-override-css-prefers-color-scheme-setting/68824350#68824350
But it requires having two copies of the variables, which is not super elegant.

Comment (Alkarex) [reply config-user.default.php]: I hope we can find a satisfying approach for { auto, yes, no, }, but if we do not implement *yes* for now, I think in any case we should have { auto, no, }, default being auto.

Comment (Frenzie) [reply config-user.default.php]: Definitely not a hold-up from my end; I was just agreeing it'd be nice to have. The SO link there looks reasonably elegant; I can't think of anything better.

Comment (math-GH) [reply config-user.default.php]: yes, this is the only solution, as far as I know.
With SCSS we could keep the maintenance effort small.

Comment (math-GH) [reply config-user.default.php]: `no` and `auto` is now implemented

**Comment by math-GH** [comment]
`No` and `auto` is now implemented as select list instead a checkbox.
Please give feedback

Todo: i18n String for `No` and `Auto`.

Out of scope of this PR: `yes` mode

**Comment by Frenzie** [comment]
Code part lgtm

**Comment by math-GH** [comment]
What could be the best i18n String?
label: `Automatic dark mode (beta)` => `Dark mode (beta)`
`no` = `No automatic dark mode`
`auto` = `Use the browser preference`? `Automatic dark mode`?

**Comment by Frenzie** [comment]
Why beta at this point? Because it only applies to one theme?

Anyway, what about:
`Appearance`/`Color scheme`

* `Browser theme`/`Browser color scheme` ("auto")
* `Light` ("no")
* `Dark` (potential future "yes")

**Comment by math-GH** [comment]
> Why beta at this point? Because it only applies to one theme?

yes. Only Origine/Origine compact will have this dark color scheme at the moment. Two themes of 13.
Each dark theme needs some testing period for fine tuning.

And we need to think about our 4 dark themes (`Alternative dark`, `dark`, `dark pink` and `Nord`). How to handle them?

That is why I would like to label it as "beta".

> 
> Anyway, what about: `Appearance`/`Color scheme`
> 
>     * `Browser theme`/`Browser color scheme` ("auto")
> 
>     * `Light` ("no")
> 
>     * `Dark` (potential future "yes")

I like this idea

**Comment by Frenzie** [comment]
> And we need to think about our 4 dark themes (Alternative dark, dark, dark pink and Nord). How to handle them?

I'd propose simply changing the screenshots, which are probably outdated and ripe for an update anyway, showing both variants with a split in the middle. Possibly with one of those fancy sliders like [this](https://levelup.gitconnected.com/how-to-create-a-before-after-image-slider-with-css-and-js-a609d9ba77bf). I don't think having a theme without variants is problematic, but if you want to give them all a light variant that's your prerogative. ;-)

**Comment by math-GH** [comment]
Let us merge this PR and collect experiences with this first dark theme.

**Comment by Alkarex** [comment]
Sorry for the delay

**Pull request merged**